src/frameworks/index.ts

```diff
@@ -538,6 +539,12 @@ ${
 
       // TODO move to templates
 
+      if (frameworksBackend && framework === "next") {
+        if (await isUsingImageOptimization(getProjectPath(), ".next")) {
+          frameworksBackend.memory = "512MiB";
+        }
+      }
```

Comment:
instead of doing it here, I’d move this to the Next.js files and return `frameworksBackend` overrides from `ɵcodegenFunctionsDirectory`, @jamesdaniels WDYT?

Reply:
I refactored this to do it in the Next.js file.